### DISCUSSION TITLE
Analog Joystick to Replace Some Switches

### DISCUSSION THREAD

**Discussion by Lester444** [kachurovskiy/nanoels#76] [Ideas]
Hello Maxim. Here's an idea on how to use an analog joystick (for example PS4) to replace the left and right switches, because for some people a joystick is much more intuitive to use compared to pressing buttons.

And by adding 2 more 470R resistors, it can also be used on the MEGA ELS to replace the big and expensive joystick. PS4 joysticks are small in size and very cheap in price.

No change to the Arduino software is needed.
![Joystick](https://user-images.githubusercontent.com/23709087/196801584-fb11f224-b159-49a8-8c8c-93ae63275fa2.png)

![Joystick2](https://user-images.githubusercontent.com/23709087/196802163-2a8698c0-845e-4905-9df1-96c7cce835f5.jpg)

**Comment by kachurovskiy** [comment]
Interesting! Few concerns:

- Is there a clear press/release sound or feedback on them?
- Does it have hysteresis issues when half-tilted?
- Is it sensitive to electromagnetic noise?

Thank you 🙏

**Reply by Lester444** [reply]
The joystick moves freely, no sound nor click. It returns to center position when released. 

Yes I did notice there is some hysteresis but it is very small.

EMI sensitivity is the same as with a regular switch, because in the software it is uses a digitalRead(xxx), not analogRead(xxx).

BTW it is not intended to be used in an analog way, but simply as a switch for logic HIGH or LOW. :)
